### DISCUSSION TITLE
Integrate vim's `diffopt` and other option control into `splice`

### DISCUSSION THREAD

**Discussion by errael** [errael/splice9#17] [Ideas]
This discussion is prompted by #16, opened by @dlamotte.
> toggle whitespace diff ... in the menu

Good idea; my first thought is _what menu_? the `HUD`? Some possibilities are a context menu, typically right-click, with options. The window would have checkboxes. If there's non boolean options, there could be text fields and/or radio buttons.

What options are important to include? I'll have to look at the code, but wrap might be inherited; don't remember. Would be nice to toggle from inside splice.

Maybe could squeeze one button into the `HUD`, which brings up a config window of some sort. Like the window which comes up when you click `Splice Commands:`. Changing the `HUD` as little as possible is  a goal (at least for now).

And probably want any things that could be done through an options window to be doable through a keyboard shortcut. Wonder what that could look like without too much polution.

In the `DirDiff` `vim` plugin there's the following which is a little cumbersome.
```
Options: 'u'=update,'x'=set excludes,'i'=set ignore,'a'=set args, 'h'=hex mode, 'w'=wrap mode
```

**Comment by dlamotte** [comment]
HUD and menu. Same thing right? Maybe it's confusing for gvim users (maybe menu refers to clickable dropdowns idk). I use just `vim` so the top options in the "HUD" in splice are the menu to me.

Regardless, I suppose I can figure out how to bind it in vim directly. Just an idea.

**Reply by errael** [reply]
> HUD and menu. Same thing right

I wasn't sure what you meant. I've seen the phrase `Menu Bar` used for something similar to the `HUD`.

I think hooking into `diffopt` is a good idea, I'd like to use that feature myself. Just wondering about the UI. A scheme that requires a new `button` in the `HUD` for each option doesn't grow very well. I've been wondering about how incorporate something for
- https://github.com/sjl/splice.vim/issues/30

Maybe one `HUD` button for `u1+u2` and one for a `diffopt` popup.

Either way, I won't be working on new features for `splice9` for a while. If you come up with a mapping, it could be useful for others if you share it in this discussion or in #16.

**Comment by errael** [comment]
With [Splice9 v1.0.0-RC3](https://github.com/errael/splice9/releases/tag/v1.0.0-RC3) there is
- `Diff Options` button brings up a dialog for setting `vim`'s `diffopt` and `wrap` options.
- `u0:  use both` is a new `splice9` command.

Still want a way to set up shortcuts for diff options.
